src/validations/rules/sar.sch

```diff
@@ -377,6 +377,34 @@
                 test="oscal:attestation[oscal:part[@name = 'authorization-statements']/oscal:prop[@ns = 'https://fedramp.gov/ns/oscal' and @name = 'recommend-authorization']]">
                 There must exist an attestation with a part containing a property with a name of 'recommend-authorization'.</sch:assert>
 
+            <sch:let
+                name="pl-2-other-than-satisfied-findings"
+                value="oscal:finding/oscal:target[@type = 'objective-id'][oscal:status ne 'satisfied'][contains(@target-id, 'pl-2')]/@target-id" />
+
+            <sch:let
+                name="pl-2-profile-objectives"
+                value="$resolved-profile-doc/oscal:catalog//oscal:part[@name eq 'objective'][ancestor::oscal:control[@id eq 'pl-2']]/@id" />
+
+            <sch:let
+                name="pl-2-not-matches-other-than-satisfied-findings"
+                value="$pl-2-other-than-satisfied-findings[not(. = $pl-2-profile-objectives)]" />
+
+            <sch:assert
+                diagnostics="multiple-control-PL-2-objectives-diagnostic"
+                fedramp:specific="true"
+                id="multiple-control-PL-2-objectives"
+                role="error"
+                test="count($pl-2-other-than-satisfied-findings) gt 0"
+                unit:override-xspec="both">Every unsatisfied PL-2 objective must have a matching resolved baseline profile catalog part.</sch:assert>
+
+            <sch:assert
+                diagnostics="multiple-control-PL-2-findings-diagnostic"
+                fedramp:specific="true"
+                id="multiple-control-PL-2-findings"
+                role="error"
+                test="count(oscal:finding[oscal:target[@type = 'objective-id' and contains(@target-id, 'pl-2')]]) lt 2">The SAR contains no more than
+                one finding for the control PL-2.</sch:assert>
```

Comment:
The `id` and `diagnostics` identifiers are the inverse of the desired outcome.
The assertion message is for the wrong context the context is `result`, not SAR.
_Unsatisfied_ findings are important, not just any finding.

```suggestion
            <sch:assert
                diagnostics="minimal-control-PL-2-findings-diagnostic"
                fedramp:specific="true"
                id="minimal-control-PL-2-findings"
                role="error"
                test="count(oscal:finding[oscal:target[@type = 'objective-id'and status/@state ne 'satisfied' and contains(@target-id, 'pl-2')]]) lt 2">Every result contains no more than
                one unsatisfied finding for the control PL-2.</sch:assert>
```